File: Projeto Didio Pizza/Didio Pizza Node/backend/src/service/produtoService.js
import * as produtoRepository from '../repository/produtoRepository.js';


export async function salvarProduto(produto) {

    if (!produto.nome) {
        throw new Error('Nome do produto é obrigatório.');
    }

    if (produto.preco <= 0) {
        throw new Error('Preço deve ser maior que zero.');
    }

    if (!['P', 'M', 'G'].includes(produto.tamanho)) {
        throw new Error('Tamanho inválido.');
    }

    let id = await produtoRepository.salvarProduto(produto);
    return id;
}


export async function listarProduto() {
    let linhas = await produtoRepository.listarProduto();
    return linhas;
}


export async function buscarProdutoPorId(id) {
    let linha = await produtoRepository.buscarProdutoPorId(id);
    return linha;
}


export async function alterarProduto(id, produto) {
    
    if (!produto.nome) {
        throw new Error('Nome do produto é obrigatório.');
    }

    if (!['P', 'M', 'G'].includes(produto.tamanho)) {
        throw new Error('Tamanho inválido.');
    }

    if (produto.preco <= 0) {
        throw new Error('Preço deve ser maior que zero.');
    }

    let linhasAfetadas = await produtoRepository.alterarProduto(id, produto);
    return linhasAfetadas;
}


export async function deletarProduto(id) {
    let linhasAfetadas = await produtoRepository.deletarProduto(id);
    return linhasAfetadas;
}


export async function atualizarFotoPizza(id, caminho) {
    let linhasAfetadas = await produtoRepository.atualizarFotoPizza(id, caminho);

    return linhasAfetadas;
} 

File: Projeto Didio Pizza/Didio Pizza Springboot/backend/src/main/java/com/pizzaria/didiopizza/service/ProdutoService.java
package com.pizzaria.didiopizza.service;

import java.util.List;
import java.util.Map;

import org.springframework.stereotype.Service;

import com.pizzaria.didiopizza.repository.ProdutoRepository;

@Service
public class ProdutoService {
    private final ProdutoRepository repo;

    public ProdutoService(ProdutoRepository repo) {
        this.repo = repo;
    }

    private double num(Object v) {
        return v == null ? 0 : Double.parseDouble(String.valueOf(v));
    }

    public Integer salvarProduto(Map<String, Object> p) {
        if (p.get("nome") == null)
            throw new RuntimeException("Nome do produto é obrigatório.");

        if (num(p.get("preco")) <= 0)
            throw new RuntimeException("Preço deve ser maior que zero.");

        if (!List.of("P", "M", "G").contains(String.valueOf(p.get("tamanho"))))
            throw new RuntimeException("Tamanho inválido.");

        return repo.salvarProduto(p);
    }

    public List<Map<String, Object>> listarProduto() {
        return repo.listarProduto();
    }

    public Map<String, Object> buscarProdutoPorId(Integer id) {
        return repo.buscarProdutoPorId(id);
    }

    public int alterarProduto(Integer id, Map<String, Object> p) {
        if (p.get("nome") == null)
            throw new RuntimeException("Nome do produto é obrigatório.");

        if (!List.of("P", "M", "G").contains(String.valueOf(p.get("tamanho"))))
            throw new RuntimeException("Tamanho inválido.");

        if (num(p.get("preco")) <= 0)
            throw new RuntimeException("Preço deve ser maior que zero.");
        
        return repo.alterarProduto(id, p);
    }

    public int deletarProduto(Integer id) {
        return repo.deletarProduto(id);
    }

    public int atualizarFotoPizza(Integer id, String caminho) {
        return repo.atualizarFotoPizza(id, caminho);
    }
}
